src/GafferScene/TweakPlug.cpp

```diff
@@ -240,70 +245,122 @@ class NumericTweak
+	if( mode == TweakPlug::Remove )
 	{
-		parameters->writable().erase( name );
+		parameters->writable().erase( parameterName );
 		return;
 	}
 
-
-	Data *parameterValue = parameters->member<Data>( name );
-	DataPtr newData = PlugAlgo::extractDataFromPlug( valuePlug() );
+	Data *parameterValue = parameters->member<Data>( parameterName );
+	DataPtr newData = PlugAlgo::extractDataFromPlug( tweakPlug->valuePlug() );
 	if( !newData )
 	{
 		throw IECore::Exception(
-			boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value plug has unsupported type \"%s\"" ) % name % valuePlug()->typeName() )
+			boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value plug has unsupported type \"%s\"" ) % tweakName % tweakPlug->valuePlug()->typeName() )
 		);
 	}
 	if( parameterValue && parameterValue->typeId() != newData->typeId() )
 	{
-		throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value of type \"%s\" does not match parameter of type \"%s\"" ) % name % parameterValue->typeName() % newData->typeName() ) );
+		throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak to \"%s\" : Value of type \"%s\" does not match parameter of type \"%s\"" ) % tweakName % parameterValue->typeName() % newData->typeName() ) );
 	}
 
-	if( mode == Replace )
+	if( mode == TweakPlug::Replace )
 	{
 		if( !parameterValue && requireExists )
 		{
-			throw IECore::Exception( boost::str( boost::format( "Cannot replace parameter \"%s\" which does not exist" ) % name ) );
+			throw IECore::Exception( boost::str( boost::format( "Cannot replace parameter \"%s\" which does not exist" ) % tweakName ) );
 		}
 
-		parameters->writable()[name] = newData;
+		parameters->writable()[parameterName] = newData;
 		return;
 	}
 
 	if( !parameterValue )
 	{
-		throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak with mode %s to \"%s\" : This parameter does not exist" ) % modeToString( mode ) % name ) );
+		throw IECore::Exception( boost::str( boost::format( "Cannot apply tweak with mode %s to \"%s\" : This parameter does not exist" ) % modeToString( mode ) % tweakName ) );
 	}
 
-	NumericTweak t( newData.get(), mode, name );
+	NumericTweak t( newData.get(), mode, tweakName );
 	dispatch( parameterValue, t );
+}
 
+} // namespace
+
+void TweakPlug::applyTweak( IECore::CompoundData *parameters, bool requireExists ) const
+{
+	const std::string name = namePlug()->getValue();
+	if( name.empty() )
+	{
+		return;
+	}
 
+	applyTweakInternal( this, name, name, parameters, requireExists );
 }
 
+void TweakPlug::applyTweaks( const Plug *tweaksPlug, IECoreScene::ShaderNetwork *shaderNetwork )
+{
+	unordered_map<InternedString, ShaderPtr> modifiedShaders;
+
+	for( TweakPlugIterator tIt( tweaksPlug ); !tIt.done(); ++tIt )
+	{
+		const TweakPlug *tweakPlug = tIt->get();
+		const std::string name = tweakPlug->namePlug()->getValue();
+		if( name.empty() )
+		{
+			continue;
+		}
+
+		ShaderNetwork::Parameter parameter;
```

Comment:
There are probably good arguments for doing it this way, but a comment just to show that I have looked through this code - I was initially confused by this parameter being constructed because I then expected it to be used as a Parameter object.  Because we only ever use the shader and name independently, I think I would find it a bit clearer if these were separate variables ( but I don't feel strongly about it ).